nuntium/tests/api/person_resource_test.py

```diff
@@ -3,40 +3,102 @@
         self.assertEqual(len(persons), Person.objects.count())  # All the instances
 
     def test_unauthorized_list_of_persons(self):
-        url = '/api/v1/person/'
+        url = "/api/v1/person/"
         response = self.api_client.get(url)
 
         self.assertHttpUnauthorized(response)
 
     def test_the_remote_url_of_the_person_points_to_its_popit_instance(self):
-        url = '/api/v1/person/'
+        url = "/api/v1/person/"
         response = self.api_client.get(url, authentication=self.get_credentials())
-        persons = self.deserialize(response)['objects']
-        self.assertEquals(persons[0]['popit_url'], persons[0]['resource_uri'])
+        persons = self.deserialize(response)["objects"]
+        self.assertEquals(persons[0]["popit_url"], persons[0]["resource_uri"])
+
+    def test_filter_list_of_persons_by_identifier(self):
+        id_count = Count("identifiers")
+        person = (
+            Person.objects.annotate(id_count=id_count).filter(id_count__gte=1).first()
+        )
+        identifier = person.identifiers.first()
+
+        url = "/api/v1/person/?identifiers__identifier=%s&identifiers__scheme=%s" % (
+            identifier.identifier,
+            identifier.scheme,
+        )
+        response = self.api_client.get(url, authentication=self.get_credentials())
+
+        self.assertValidJSONResponse(response)
+
+        persons = self.deserialize(response)["objects"]
+        self.assertEqual(1, len(persons))
+        self.assertEqual(person.id, persons[0]["id"])
+
+    def test_filter_list_of_persons_by_has_contacts(self):
+        url = "/api/v1/person/?has_contacts=True"
+        response = self.api_client.get(url, authentication=self.get_credentials())
+
+        self.assertValidJSONResponse(response)
+
+        persons = self.deserialize(response)["objects"]
+        self.assertEqual(len(persons), Person.objects.has_contacts().count())
+
+    def test_filter_list_of_persons_by_does_not_have_contacts(self):
+        url = "/api/v1/person/?has_contacts=False"
+        response = self.api_client.get(url, authentication=self.get_credentials())
+
+        self.assertValidJSONResponse(response)
+
+        persons = self.deserialize(response)["objects"]
+        self.assertEqual(len(persons), Person.objects.doesnt_have_contacts().count())
+
+    def test_filter_invalid_has_contacts_value(self):
+        url = "/api/v1/person/?has_contacts=something-else"
+        response = self.api_client.get(url, authentication=self.get_credentials())
+
+        self.assertHttpBadRequest(response)
+
+    def test_filter_list_of_persons_by_instance(self):
+        url = "/api/v1/person/?instance_id=%d" % self.writeitinstance.id
+        response = self.api_client.get(url, authentication=self.get_credentials())
+
+        self.assertValidJSONResponse(response)
+
+        persons = self.deserialize(response)["objects"]
+        self.assertEqual(len(persons), self.writeitinstance.persons.count())
```

Comment:
I'm nervous of a change to setUp invisibly breaking the tests based on the ORM calls - could you include that hardcoded number assertion in each?